core/lib/instrument.c

```diff
@@ -6478,8 +6478,13 @@ dr_get_mcontext_priv(dcontext_t *dcontext, dr_mcontext_t *dmc, priv_mcontext_t *
          * when most old clients have been converted, remove this (we'll
          * still return false)
          */
-        CLIENT_ASSERT(dmc->size == sizeof(dr_mcontext_t),
-                      "dr_mcontext_t.size field not set properly");
+        CLIENT_ASSERT(
+            dmc->size == sizeof(dr_mcontext_t) ||
+                /* Opmask storage has been added for AVX-512 (xref i#1312).
+                 * An older client's mcontext may be filled in w/o the new structure.
+                 */
+                dmc->size == sizeof(dr_mcontext_t) - sizeof(dr_opmask_t),
+            "dr_mcontext_t.size field not set properly");
```

Comment:
I would just remove this assert as it's going to get un-maintainable with a long list of valid sizes.  Ditto below.

Reply:
Done.